src/main/java/leetCode/TicketPath.java

```diff
@@ -4,7 +4,37 @@
 
 public class TicketPath {
     private static HashMap<String, LinkedHashSet<String>> flights;
+    private static HashMap<String, String> flights2;
 
+//    No use of helper functions
+    public static String ticketPathDecoderV2(Stack<Ticket> stackOfTickets){
+        flights2 = new HashMap<>();
+        ArrayList<String> arrivals  = new ArrayList<>();
+
+        for(int i = stackOfTickets.size(); i > 0 ; i--){
+            Ticket ttp = stackOfTickets.pop();
+            arrivals.add(ttp.arrive);
+            flights2.put(ttp.depart,ttp.arrive);
+        }
+//        get all the flight departures and compare to the arrivals
+        for(String city : flights2.keySet()) {

```

Comment:
@ahrenswett can you add tests to it to make sure this works?

Reply:
yes there is a situation it will not work in. doubles of any airport.